src/main/scala/com/mozilla/telemetry/mozdata/MozData.scala

```diff
@@ -0,0 +1,335 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package com.mozilla.telemetry.mozdata
+
+import java.util.UUID.randomUUID
+
+import com.mozilla.telemetry.heka.{Dataset, Message}
+import com.mozilla.telemetry.mozdata.Utils._
+import com.mozilla.telemetry.utils.S3Store
+import org.apache.log4j.Logger
+import org.apache.spark.rdd.RDD
+import org.json4s.jackson.Serialization.{read, write}
+import org.json4s.{DefaultFormats, Formats}
+import org.apache.spark.sql._
+import scalaj.http.Http
+
+import scala.collection.immutable.ListMap
+import scala.io.Source
+
+/** A consistent API for accessing Mozilla data that reports usage to Mozilla
+  *
+  * @param spark spark session used to access data
+  * @param adHocTablesDir filesystem location of ad hoc tables
+  * @param globalTablesDir filesystem location of global tables
+  * @param defaultMetadataUpdateMethods default list of methods to use when
+  *                                     updating table metadata
+  */
+class MozData(spark: SparkSession, adHocTablesDir: String, globalTablesDir: String,
+              defaultMetadataUpdateMethods: List[String],
+              readConfig: Function[DataFrameReader,DataFrameReader],
+              writeConfig: Function[DataFrameWriter[Row],DataFrameWriter[Row]],
+              telemetryUrl: Option[String]){
+  val apiVersion: String = Source.fromFile("VERSION").mkString
+  val logger: Logger = Logger.getLogger(classOf[MozData])
+
+  /** Report an interaction with this api */
+  private def log(action: String, event: Map[String, Option[String]] = Map()): Unit = {
+    implicit val formats: Formats = DefaultFormats
+    val ping: String = write(ListMap((event.collect {
+      case (k, Some(v)) => k -> v
+    } + ("apiVersion"->apiVersion)).toList.sorted:_*))
+    if (telemetryUrl.isDefined) {
+      val url = s"${telemetryUrl.get}/submit/mozza/mozdata.$action/1/${randomUUID.toString}"
```

Comment:
This is my bad - I recommended putting the schemas for these messages in with the existing `mozza` namespace. I think it might make more sense to have a `mozdata` namespace with various schemas for different message types.

Reply:
I'm not sure we want different tables for every message type, that seems like a lot of work on updates (e.g. adding new messages, adding new fields to existing messages, etc.). My preference would be one new mozdata ping, with the top-level fields being shared among all logs (e.g. `apiVersion`, `userEmail`, `sparkVersion`, etc.) and then an extras MAP field for log-specific fields.

Reply:
The url format is `submit/namespace/doctype/version`, so this structure is doing one-table-per-message-type (namespace is `mozza`, doctype+table name is `mozdata.$action`).

Reply:
I'll butt out for the discussion of one vs. many tables though :)